server/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java

```diff
@@ -427,8 +427,14 @@ public IndexAnalyzers build(IndexSettings indexSettings,
         Map<String, NamedAnalyzer> normalizers = new HashMap<>();
         Map<String, NamedAnalyzer> whitespaceNormalizers = new HashMap<>();
         for (Map.Entry<String, AnalyzerProvider<?>> entry : analyzerProviders.entrySet()) {
-            processAnalyzerFactory(indexSettings, entry.getKey(), entry.getValue(), analyzers,
-                tokenFilterFactoryFactories, charFilterFactoryFactories, tokenizerFactoryFactories);
+            analyzers.merge(entry.getKey(), produceAnalyzer(entry.getKey(), entry.getValue(), tokenFilterFactoryFactories,
+                    charFilterFactoryFactories, tokenizerFactoryFactories), (k, v) -> {
+                        throw new IllegalStateException("already registered analyzer with name: " + entry.getKey());
+                    });
+            final String analyzerAliasKey = "index.analysis.analyzer." + entry.getKey() + ".alias";
```

Comment:
The original code had this comment: // TODO: remove alias support completely when we no longer support pre 5.0 indices
Should we remove this part as we don't support pre 5.0 indexes any more

Reply:
Thanks, I was wondering the same. I understood the comment to mean we want to remove the alias support (which I think we have already) but not the error that prevents these settings to be silently ignored going forward. But I will take another look and maybe consult with sb. else before proceeding to merge this.

Reply:
+1 to remove, this is not supported nor documented since version 5.